aptos-move/framework/aptos-stdlib/sources/data_structures/ordered_map.move

```diff
@@ -0,0 +1,529 @@
+/// This module provides an implementation for an ordered map.
+///
+/// Keys point to values, and each key in the map must be unique.
+///
+/// Currently, one implementation is provided, backed by a single sorted vector.
+///
+/// That means that keys can be found within O(log N) time.
+/// Adds and removals take O(N) time, but the constant factor is small,
+/// as it does only O(log N) comparisons, and does efficient mem-copy with vector operations.
+///
+/// Additionally, it provides a way to lookup and iterate over sorted keys, making range query
+/// take O(log N + R) time (where R is number of elements in the range).
+///
+/// Most methods operate with OrderedMap being `self`.
+/// All methods that start with iter_*, operate with Iterator being `self`.
+///
+/// Uses cmp::compare for ordering, which compares primitive types natively, and uses common
+/// lexicographical sorting for complex types.
+module aptos_std::ordered_map {
+    // friend aptos_std::big_ordered_map;
```

Comment:
？

Reply:
will be uncommented in the PR with big_ordered_map